src/lib.rs

```diff
@@ -214,15 +229,28 @@ impl Uart {
 
     /// Reads and returns a pending byte, or `None` if nothing has been
     /// received.
-    pub fn read_byte(&mut self) -> Option<u8> {
+    pub fn read_byte(&mut self) -> Result<Option<u8>, Error> {
         if self.flags().contains(Flags::RXFE) {
-            None
+            Ok(None)
         } else {
             // SAFETY: self.registers points to the control registers of a PL011 device which is
             // appropriately mapped, as promised by the caller of `Uart::new`.
             let data = unsafe { addr_of!((*self.registers).dr).read_volatile() };
-            // TODO: Check for error conditions in bits 8-11.
-            Some(data as u8)
+            // Check for errors in receive status register.
+            let error_status = self.receive_status();
```

Comment:
Can we get the error information from bits 8-11 of the data register rather than doing a separate register read here?